packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/ClusterManagersController.m

```diff
@@ -0,0 +1,157 @@
+
+@property(strong, nonatomic) NSMutableDictionary *clusterManagerIdToManagers;
+@property(strong, nonatomic) FlutterMethodChannel *methodChannel;
+@property(weak, nonatomic) GMSMapView *mapView;
+
+@end
+
+@implementation FLTClusterManagersController
+- (instancetype)init:(FlutterMethodChannel *)methodChannel mapView:(GMSMapView *)mapView {
+  self = [super init];
+  if (self) {
+    _methodChannel = methodChannel;
+    _mapView = mapView;
+    _clusterManagerIdToManagers = [[NSMutableDictionary alloc] init];
+  }
+  return self;
+}
+
+- (void)addClusterManagers:(NSArray *)clusterManagersToAdd {
+  for (NSDictionary *clusterManager in clusterManagersToAdd) {
+    NSString *identifier = clusterManager[@"clusterManagerId"];
+    id<GMUClusterAlgorithm> algorithm = [[GMUNonHierarchicalDistanceBasedAlgorithm alloc] init];
+    id<GMUClusterIconGenerator> iconGenerator = [[GMUDefaultClusterIconGenerator alloc] init];
+    id<GMUClusterRenderer> renderer =
+        [[GMUDefaultClusterRenderer alloc] initWithMapView:_mapView
+                                      clusterIconGenerator:iconGenerator];
+    GMUClusterManager *clusterManager = [[GMUClusterManager alloc] initWithMap:_mapView
+                                                                     algorithm:algorithm
+                                                                      renderer:renderer];
+    self.clusterManagerIdToManagers[identifier] = clusterManager;
+  }
+}
+
+- (void)removeClusterManagers:(NSArray *)identifiers {
+  for (NSString *identifier in identifiers) {
+    GMUClusterManager *clusterManager = [self.clusterManagerIdToManagers objectForKey:identifier];
+    if (!clusterManager) {
+      continue;
+    }
+    [clusterManager clearItems];
+    [self.clusterManagerIdToManagers removeObjectForKey:identifier];
+  }
+}
+
+- (void)addItem:(GMSMarker *)marker clusterManagerId:(NSString *)clusterManagerId {
+  GMUClusterManager *clusterManager =
+      [self.clusterManagerIdToManagers objectForKey:clusterManagerId];
+  if (marker && clusterManager != (id)[NSNull null]) {
+    [clusterManager addItem:(id<GMUClusterItem>)marker];
+  }
+}
+
+- (void)removeItem:(GMSMarker *)marker clusterManagerId:(NSArray *)clusterManagerId {
+  GMUClusterManager *clusterManager =
+      [self.clusterManagerIdToManagers objectForKey:clusterManagerId];
+  if (marker && clusterManager != (id)[NSNull null]) {
+    [clusterManager removeItem:(id<GMUClusterItem>)marker];
+  }
+}
+
+- (void)clusterAll {
+  for (GMUClusterManager *clusterManager in [self.clusterManagerIdToManagers allValues]) {
+    [clusterManager cluster];
+  }
+}
+
+- (void)getClustersWithIdentifier:(NSString *)identifier result:(FlutterResult)result {
+  GMUClusterManager *clusterManager = [self.clusterManagerIdToManagers objectForKey:identifier];
+  if (clusterManager) {
+    NSMutableArray *response = [[NSMutableArray alloc] init];
+
+    NSUInteger integralZoom = (NSUInteger)floorf(_mapView.camera.zoom + 0.5f);
```

Comment:
Why do we need an integralZoom? Could you add a comment to explain?

Reply:
Added ref comment about this.